docs/addons/nats/tls/index.md

```diff
@@ -0,0 +1,741 @@
+
+Configuration:
+
+             Subjects: ORDERS.*
+     Acknowledgements: true
+            Retention: File - Limits
+             Replicas: 1
+       Discard Policy: Old
+     Duplicate Window: 2m0s
+     Maximum Messages: unlimited
+        Maximum Bytes: unlimited
+          Maximum Age: 1y0d0h0m0s
+ Maximum Message Size: unlimited
+    Maximum Consumers: unlimited
+
+
+Cluster Information:
+
+                 Name: nats
+               Leader: nats-0
+
+State:
+
+             Messages: 0
+                Bytes: 0 B
+             FirstSeq: 0
+              LastSeq: 0
+     Active Consumers: 0
+
+
+# Verify that the stream has been created successfully
+sample-nats-box-785f8458d7-wtnfx:~#  nats stream ls
+Streams:
+
+	ORDERS
+	
+# Lets add some messages to the stream "ORDERS"
+sample-nats-box-785f8458d7-wtnfx:~# nats pub ORDERS.scratch hello
+08:55:39 Published 5 bytes to "ORDERS.scratch"
+
+# Add another message
+sample-nats-box-785f8458d7-wtnfx:~# nats pub ORDERS.scratch world
+08:56:11 Published 5 bytes to "ORDERS.scratch"
+
+# Verify that the messages have been published to the stream successfully
+sample-nats-box-785f8458d7-wtnfx:~# nats stream info ORDERS
+Information for Stream ORDERS created 2021-09-03T07:12:07Z
+
+Configuration:
+
+             Subjects: ORDERS.*
+     Acknowledgements: true
+            Retention: File - Limits
+             Replicas: 1
+       Discard Policy: Old
+     Duplicate Window: 2m0s
+     Maximum Messages: unlimited
+        Maximum Bytes: unlimited
+          Maximum Age: 1y0d0h0m0s
+ Maximum Message Size: unlimited
+    Maximum Consumers: unlimited
+
+
+Cluster Information:
+
+                 Name: nats
+               Leader: nats-0
+
+State:
+
+             Messages: 2
+                Bytes: 98 B
+             FirstSeq: 1 @ 2021-09-03T08:55:39 UTC
+              LastSeq: 2 @ 2021-09-03T08:56:11 UTC
+     Active Consumers: 0
+
+sample-nats-box-785f8458d7-wtnfx:~# exit
+```
+
+We have successfully deployed a NATS cluster, created a stream and publish some messages into the stream. In the subsequent sections, we are going to backup this sample data using Stash.
+
+## Prepare for Backup
+
+In this section, we are going to prepare the necessary resources (i.e. connection information, backend information, etc.) before backup.
+
+### Ensure NATS Addon
+
+When you install Stash Enterprise version, it will automatically install all the official addons. Make sure that NATS addon was installed properly using the following command.
+
+```bash
+❯ kubectl get tasks.stash.appscode.com | grep nats
+nats-backup-2.4.0            24m
+nats-restore-2.4.0           24m
+```
+
+This addon should be able to take backup of the NATS streams with matching major versions as discussed in [Addon Version Compatibility](/docs/addons/nats/README.md#addon-version-compatibility).
+
+### Create Secret
+
+ Lets create a secret with access credentials.  Below is the YAML of `Secret` object we are going to create.
+
+```yaml
```

Comment:
Don't create this secret manually. Use cert-manager to issue client certificates